File: gin-vue-admin/web/src/api/ffMaterial.js
import service from '@/utils/request'

export const createMaterial = (data) => {
  return service({
    url: '/ffProduct/createMaterial',
    method: 'post',
    data
  })
}

export const updateMaterial = (data) => {
  return service({
    url: '/ffProduct/updateMaterial',
    method: 'put',
    data
  })
}

export const deleteMaterial = (data) => {
  return service({
    url: '/ffProduct/deleteMaterial',
    method: 'delete',
    data
  })
}

export const getMaterialList = (params) => {
  return service({
    url: '/ffProduct/getMaterialList',
    method: 'get',
    params
  })
}

File: gin-vue-admin/server/router/ffproduction/ff_material.go
package ffproduction

import (
	"gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type MaterialRouter struct{}

func (e *MaterialRouter) InitMaterialRouter(Router *gin.RouterGroup) {
	materialRouter := Router.Group("ffProduct").Use(middleware.OperationRecord())
	materialRouterWithoutRecord := Router.Group("ffProduct")
	{
		materialRouter.POST("createMaterial", materialApi.CreateMaterial)
		materialRouter.PUT("updateMaterial", materialApi.UpdateMaterial)
		materialRouter.DELETE("deleteMaterial", materialApi.DeleteMaterial)
	}
	{
		materialRouterWithoutRecord.GET("getMaterialList", materialApi.GetMaterialList)
	}
}
